src/components/phone-input/PhoneInput.tsx

```diff
@@ -0,0 +1,139 @@
+import {useState} from 'react';
+import {PhoneInputSectionItem, Section} from '@atb/components/sections';
+import {ActivityIndicator, StyleProp, View, ViewStyle} from 'react-native';
+import {LoginTexts, useTranslation} from '@atb/translations';
+import {StyleSheet, useTheme} from '@atb/theme';
+import {PhoneSignInErrorCode} from '@atb/auth';
+import {StaticColorByType, getStaticColor} from '@atb/theme/colors';
+import {MessageInfoBox} from '@atb/components/message-info-box';
+import {Button} from '@atb/components/button';
+import phone from 'phone';
+import {SvgProps} from 'react-native-svg';
+
+type Props = {
+  style?: StyleProp<ViewStyle>;
+  submitButtonText: string;
+  submitButtonTestId: string;
+  onNextPromise: (
+    number: string,
+    forceResend?: boolean,
+  ) => Promise<PhoneSignInErrorCode | undefined>;
+  onNextAction: (number: string) => void;
+  rightIcon?: (props: SvgProps) => JSX.Element;
+};
+
+const themeColor: StaticColorByType<'background'> = 'background_accent_0';
+
+export const PhoneInput = ({
+  style,
+  submitButtonText,
+  submitButtonTestId,
+  onNextPromise,
+  onNextAction,
```

Comment:
Should either choose the term "submit" or "next" for the fields. I think "submit" is better.

Reply:
Got it, thanks for the suggestion, I was just using the existing naming but `onSubmitPromise` and `onSubmitAction` sounds better.

Reply:
Updated!